internal/daemon/daemon.go

```diff
@@ -501,6 +501,23 @@ func (d *Daemon) ClusterCert() *shared.CertInfo {
 	return d.clusterCert
 }
 
+// ReloadClusterCert reloads the cluster keypair from the state directory.
+func (d *Daemon) ReloadClusterCert() error {
+	d.clusterMu.Lock()
+	defer d.clusterMu.Unlock()
+
+	clusterCert, err := util.LoadClusterCert(d.os.StateDir)
+	if err != nil {
+		return err
+	}
+
+	d.clusterCert = clusterCert
+	d.endpoints.UpdateTLS(clusterCert)
+	d.db.ClusterCert = clusterCert
```

Comment:
Maybe instead of exporting `db.ClusterCert` we can have a function `func (db *DB) UpdateClusterCert()` to keep things separated?